### PR TITLE
DataFactory dependency update

### PR DESCRIPTION
**Pull request by Frey-Wang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repository and upgrade the version in the pull request but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
DataFactory

File: src/datafactory/HISTORY.rst [hunk 1]
```diff
@@ -2,6 +2,9 @@
 
 Release History
 ===============
+0.6.0
++++++
+* dependency module update
```
Comment (evelyn-ys) [inline src/datafactory/HISTORY.rst:7]: Please describe which bug is fixed in this version

Comment (Frey-Wang) [reply src/datafactory/HISTORY.rst]: Updated.

File: src/datafactory/setup.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 from setuptools import setup, find_packages
 
 # HISTORY.rst entry.
-VERSION = '0.1.0'
+VERSION = '0.6.0'
```
Comment (evelyn-ys) [inline src/datafactory/setup.py:13]: This version is overwritten in https://github.com/Azure/azure-cli-extensions/blob/main/src/datafactory/azext_datafactory/manual/version.py#L11

Can you please update there instead?

Comment (Frey-Wang) [reply src/datafactory/setup.py]: Updated

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**